examples/sui-move/color_object/sources/example.move

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module color_object::example {
+    use sui::object::{Self, UID};
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+
+    struct ColorObject has key, store {
+        id: UID,
+        red: u8,
+        green: u8,
+        blue: u8,
+    }
+
+    // === Functions covered in Chapter 1 ===
+
+    public fun new(
+        red: u8,
+        green: u8,
+        blue: u8,
+        ctx: &mut TxContext,
+    ): ColorObject {
+        ColorObject { id: object::new(ctx), red, green, blue }
+    }
+
+    public fun get_color(self: &ColorObject): (u8, u8, u8) {
+        (self.red, self.green, self.blue)
+    }
+
+    // === Functions covered in Chapter 2 ===
+
+    /// Copies the values of `from` into `into`.
+    public fun copy_into(from: &ColorObject, into: &mut ColorObject) {
```

Comment:
nit: Would it make sense to switch the params ordering now, to make it compatible with the receiver syntax in the future? (if I've understood it properly).

Reply:
I think this is still compatible with receiver syntax, i.e. `from.copy_into(&mut into)` still makes sense.

Reply:
@amnn True! It was mostly me thinking of the receiver as the one being mutated 😄

Reply:
Ah, that's not generally true (i.e. not a recommended style to always have the mutated object be the receiver) because it would interfere with other patterns.  In this case, if we switched the order of parameters, we would also need to rename the function to `copy_from` to make the receiver syntax read correctly.  I think either of those functions are fine though, so I'll leave this as-is.